src/main/java/io/odpf/firehose/filter/json/JsonFilter.java

```diff
@@ -0,0 +1,120 @@
+package io.odpf.firehose.filter.json;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.protobuf.GeneratedMessageV3;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.util.JsonFormat;
+import com.networknt.schema.JsonSchema;
+import com.networknt.schema.JsonSchemaFactory;
+import com.networknt.schema.SpecVersion;
+import com.networknt.schema.ValidationMessage;
+import io.odpf.firehose.config.FilterConfig;
+import io.odpf.firehose.consumer.Message;
+import io.odpf.firehose.filter.Filter;
+import io.odpf.firehose.filter.FilterException;
+import io.odpf.firehose.metrics.Instrumentation;
+import org.apache.commons.lang3.reflect.MethodUtils;
+
+import java.lang.reflect.InvocationTargetException;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import static io.odpf.firehose.config.enums.FilterDataSourceType.KEY;
+import static io.odpf.firehose.config.enums.FilterDataSourceType.NONE;
+
+/**
+ * JSON-based filter to filter protobuf/JSON messages based on rules
+ * defined in a JSON Schema string.
+ */
+public class JsonFilter implements Filter {
+
+    private final ObjectMapper objectMapper;
+    private final Instrumentation instrumentation;
+    private final JsonSchema schema;
+    private final JsonFormat.Printer jsonPrinter;
+    private final FilterConfig filterConfig;
+
+    /**
+     * Instantiates a new Json filter.
+     *
+     * @param filterConfig    the consumer config
+     * @param instrumentation the instrumentation
+     */
+    public JsonFilter(FilterConfig filterConfig, Instrumentation instrumentation) {
+        objectMapper = new ObjectMapper();
+        this.filterConfig = filterConfig;
+        JsonSchemaFactory schemaFactory = JsonSchemaFactory.getInstance(SpecVersion.VersionFlag.V7);
+        jsonPrinter = JsonFormat.printer().preservingProtoFieldNames();
+        this.instrumentation = instrumentation;
+        if (filterConfig.getFilterJsonDataSource() != NONE) {
+            schema = schemaFactory.getSchema(filterConfig.getFilterJsonSchema());
+        } else {
+            schema = null;
+        }
+    }
+
+    /**
+     * method to filter the EsbMessages.
+     *
+     * @param messages the json/protobuf records in binary format that are wrapped in {@link Message}
+     * @return the list of filtered Messages
+     * @throws FilterException the filter exception
+     */
+    @Override
+    public List<Message> filter(List<Message> messages) throws FilterException {
+        if (filterConfig.getFilterJsonDataSource() == NONE) {
```

Comment:
let's have a explicit config for filter_enable and remove the NONE type..
we can also change in previous json filter class. 
```
if (!filterConfig.isFilterEnable) return messages;
```

Reply:
OR have a NONE type in FILTER_ENGINE type.. if it's not added then we can have NoOp implementation of the filter  interface and we can pass that class.

Reply:
fixed